### PR TITLE
feat: add latest height query

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes CFE-485

### CODE REVIEW

File: src/lib/app-provider/env.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export enum CELATONE_QUERY_KEYS {
   // BLOCK
   BLOCKS = "CELATONE_QUERY_BLOCKS",
   BLOCK_DATA = "CELATONE_QUERY_BLOCK_DATA",
+  BLOCK_LATEST_HEIGHT = "CELATONE_QUERY_BLOCK_LATEST_HEIGHT",
```
Comment (songwongtp) [inline src/lib/app-provider/env.ts:31]: BLOCK_LATEST_HEIGHT_LCD = "CELATONE_QUERY_BLOCK_LATEST_HEIGHT_LCD"

File: src/lib/services/block/index.ts [hunk 2]
```diff
@@ -33,3 +39,18 @@ export const useBlockData = (height: number, enabled = true) => {
     }
   );
 };
+
+export const useLatestBlock = () => {
```
Comment (songwongtp) [inline src/lib/services/block/index.ts:43]: useLatestBlockLcd

**Review by songwongtp — APPROVED** [review]

File: src/lib/layout/InformationFooter.tsx [hunk 3]
```diff
@@ -47,9 +55,9 @@ export const InformationFooter = () => {
         <Flex gap={1} py={1} px={2} align="center">
           <CustomIcon name="block" color="gray.600" boxSize={3} />
           <Text variant="body3" color="text.dark">
```
[the remark this answers is not in the document]
Comment (evilpeach) [reply src/lib/layout/InformationFooter.tsx]: Sure!

File: src/lib/services/block/index.ts [hunk 4]
```diff
@@ -33,3 +39,18 @@ export const useBlockData = (height: number, enabled = true) => {
     }
   );
 };
+
+export const useLatestBlockLcd = () => {
+  const { currentChainId } = useCelatoneApp();
```
Comment (songwongtp) [inline src/lib/services/block/index.ts:44]: don't think we need current chain id?

File: src/lib/services/block/lcd.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+
+import { zBlockLcd } from "../types";
+import { parseWithError } from "lib/utils";
+
+export const getLatestBlockLcd = async (endpoint: string) => {
+  return axios
```
Comment (songwongtp) [inline src/lib/services/block/lcd.ts:7]: remove return

**Review by songwongtp — APPROVED** [review]

File: src/lib/services/block/index.ts (hunk 4 above)
Comment (evilpeach) [reply src/lib/services/block/index.ts:44]: yes

**Pull request merged**